core-api/src/telemetry.rs

```diff
@@ -82,6 +86,9 @@ pub enum MetricsExporter {
     Otel(OtelCollectorOptions),
     /// Expose metrics directly via an embedded http server bound to the provided address.
     Prometheus(SocketAddr),
+    /// Export metrics calls to lang where it can inject them into whatever metrics system the
+    /// user likes
+    Lang(Arc<dyn CoreMeter>),
 }
```

Comment:
I'm not sure if we need to rework this, but we need to be able to access the `CoreMeter` for the OTel/Prom impls too. Langs need to be able to record custom metrics on the existing metric impls. With the API as it is today, even if the OTel/Prom impls are made public, langs will never call the `Otel` and `Prometheus` enums again on `MetricsExporter`.

Maybe we can remove `MetricsExporter` and just make `CoreMeter` the trait accepted in Telemetry and have functions to create `CoreMeter` for OTel/Prom (and of course lang can just impl `CoreMeter` themselves). Otherwise, can we have some other way to access the `CoreMeter`? I wonder if we should also make standard tag names available.

Reply:
Standard tag names available as constants? Yeah, that makes sense to me.

I can see getting rid of `MetricsExporter` and making `CoreMeter` the only thing, and having a way to get one for Otel & Prom. @mjameswh What do you think? I will make this work if we all like it.

Reply:
:+1: Yeah, anything that will let us both (and separately) create custom meter _and_ create custom metrics on whatever meter is created will work great!

Reply:
I agree: the `MetricsExporter` enum doesn't add much value. Not sure however this is blocker for lang->core custom metrics, as we can get the configured `CoreMeter` object from `CoreRuntime.metric_meter`. But still, just to keep things simpler, we should probably remove the useless enum.

One minor thing however is that both the Prometheus and the Otel metric initialization code currently depends on the Tokio `runtime` object; seems like we use that to run the Prometheus server on a distinct thread. Moving that initialization

Reply:
Done